Tasks/PackerBuild/task.json

```diff
@@ -117,17 +117,25 @@
         {
             "target": "location",
             "endpointId": "$(ConnectedServiceName)",
-            "dataSourceName": "AzureLocations"
+            "dataSourceName": "AzureLocations2",
+            "resultTemplate": "{ \"Value\" : \"{{{name}}}\", \"DisplayValue\" : \"{{{displayName}}}\" }"
         },
         {
-            "target": "azureStorageAccount",
+            "target": "storageAccountName",
             "endpointId": "$(ConnectedServiceName)",
-            "dataSourceName": "AzureStorageAccountRM"
+            "dataSourceName": "AzureRMStorageAccountByLocation",
+            "parameters": {
+                "location": "$(location)"
+            }
         },
         {
             "target": "azureResourceGroup",
             "endpointId": "$(ConnectedServiceName)",
-            "dataSourceName": "AzureResourceGroups"
+            "dataSourceName": "AzureRMStorageAccountIdByName",
```

Comment:
as per discussion, this will extract resourceGroup name from selected storage account's id field.